src/MacroTools/ObjectiveSystem/Objectives/UnitBased/ObjectiveAnyEnemyUnitInRects.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.QuestSystem;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.ObjectiveSystem.Objectives.UnitBased
+{
+  /// <summary>
+  /// Completed when an eligible player moves a unit into the specified <see cref="Rectangle"/>.
```

Comment:
Doesn't look quite correct. Probably more like:
```suggestion
  /// Completed when an enemy of an eligible Faction moves a unit into the specified <see cref="Rectangle"/>.
```